src/symbols_comp.c

```diff
@@ -1,4 +1,36 @@
-#include "symbols.h"
+#include "symbols_comp.h"
+
+/*******************************************************************************
+ * Forward static declarations
+ ******************************************************************************/
+
+static void comp_syms_clean_one_entry(void*);
+static int comp_syms_compare(const void*, const void*);
+static void comp_syms_print_one(void*);
+
+/*******************************************************************************
+ * Helper functions
+ ******************************************************************************/
+
+static void
+comp_syms_clean_one_entry(void *sym)
+{
+    free(sym);
+}
+
+static int
+comp_syms_compare(const void* arg, const void* item)
```

Comment:
Is there a reason why several of these functions take `void *` rather than a specific type? It seems that they all end up casting the `void *` to a more specific type?

Reply:
Yes, that's just how the API works unfortunately. For search [1], for example, a `tommy_search_func` takes `(const void*, const void*)` as arguments.

[1] https://www.tommyds.it/doc/tommyhashtbl_8h.html#a9a2455500c2ed927005ca7d222aa60ff
[2] https://www.tommyds.it/doc/tommytypes_8h.html#a9008971cd09a3f9c4f9da45cffb4627d

Reply:
You *might* find it easier to cast the function type itself at the interface to tommy? But that might not end up being a win...

Reply:
You mean instead of
```
    lib_symbol* found = tommy_hashtable_search(list, lib_syms_compare,
            to_find, hashtable_hash(to_find));
```
, replace `to_find` with something like `(int)(*)(const void*, const void*) to_find`? I think that looks a bit ugly, compared to the current version, and a cursory google search [1] shows it *might* be UB (I've not perused the standard, but reading the answer, I personally infer it would be).

[1] https://stackoverflow.com/a/559671